### PR TITLE
Feature/add book search and sort

### PR DESCRIPTION
**Pull request by khalidkanaan**

### CODE REVIEW

File: src/main/resources/static/stylesheets/common.css [hunk 1]
```diff
@@ -8,18 +8,19 @@
 #home-button{
     background-color: transparent;
     border: none;
-    margin-left: 3%;
-    margin-top: 2%;
-    width: 8%;
+    margin-left: 20px;
```
Comment (Brandon-999) [inline src/main/resources/static/stylesheets/common.css:11]: Why absolute values instead of relative?

**Comment by Brandon-999** [comment]
This needs a description. Can you provide a bullet point list of what changed?

**Review by Brandon-999 — DISMISSED** [review]
Looks fine to merge from what I can tell. I'll approve for now in case you just want to get it merged.

File: src/main/resources/static/stylesheets/common.css (hunk 1 above)
Comment (khalidkanaan) [reply src/main/resources/static/stylesheets/common.css]: Fixed.

**Review by Brandon-999 — APPROVED** [review]
looks great

**Comment by khalidkanaan** [comment]
> This needs a description. Can you provide a bullet point list of what changed?

This PR looks like a lot since we populate our book store with a few more books (see [https://github.com/Brandon-999/4806_project/pull/90/files#diff-2ba43fb809f81623ef3d3ac552d2d7b587b49bff2c4ebaa9cc89997578edd1ba,](BookService.java)) here's a summary of what we've done:
- For the search and sort functionalities we've decided to use the datatables JQuery plugin to organize all of our frontpage books in a table that can perform searches using JQuery instead of creating backend functions that query our MySQL database. This will have a positive impact on performance and make our searching and sorting much simpler.
- In common.js you'll see that we've replaced our addBookButton() function with generateBookTable() function which organizes and displays all the homepage books in our searchable and sortable table. This scales much better than what we had before with the clickable bookcover images that do not provide any information to the user unless they visit the book page.
- In [https://github.com/Brandon-999/4806_project/pull/90/commits/16ddec47eb6f50b5c719db3cb97d2ee97fdbf0b8#diff-0a219aeecc08edb51017bd52c2556d6d9def1dc1adaefa41e99ab3846eef15ca](BookModel.java), we've added a genre attribute and increased the description length of our books to varchar(1000) to allow us to store longer book descriptions in our database.
- In [https://github.com/Brandon-999/4806_project/pull/90/files#diff-e4749505434117d43fcb9ebfb3387ab73831a3d60e50529e0d9299413999c98f](Order.js), we've cleaned up the code a bit to use more meaningful variable names.

**Pull request merged**